libcontainer/process_linux.go

```diff
@@ -144,6 +145,14 @@ func (p *setnsProcess) start() (retErr error) {
 			}
 		}
 	}
+	// Reset the inherited cpu affinity. Old kernels do that automatically, but
+	// new kernels remember the affinity that was set before the cgroup move.
+	// This is undesirable, because it inherits the systemd affinity when the container
+	// should really move to the container space cpus.
+	if err = unix.SchedSetaffinity(p.pid(), &unix.CPUSet{}); err != nil && !errors.Is(err, syscall.EINVAL) && !errors.Is(err, syscall.ENODEV) {
```

Comment:
1. Please use `unix.EINVAL` and `unix.ENODEV`.
2. No need for `errors.Is` -- you can compare those values directly (i.e. `err != unix.EINVAL` etc.).
3. Add a reference to https://issues.redhat.com/browse/OCPBUGS-15102 to the commit message.
4. This code path is just for `runc exec`. If you need this to be done for `runc start/run` as well, you need to patch `(p *initProcess) start()` as well.

NB: similar crun PR: https://github.com/containers/crun/pull/1315

Reply:
updated!

Reply:
should I override linter https://github.com/opencontainers/runc/actions/runs/6342217795?

Reply:
Ah! Using `err :=` (i.e. instantiating a new variable instead of reusing an existing one) should fix this.